airbyte-connector-builder-server/connector_builder/impl/default_api.py

```diff
@@ -2,21 +2,344 @@
+from connector_builder.generated.models.http_request import HttpRequest
+from connector_builder.generated.models.http_response import HttpResponse
+from connector_builder.generated.models.known_exception_info import KnownExceptionInfo
 from connector_builder.generated.models.stream_read import StreamRead
+from connector_builder.generated.models.stream_read_pages import StreamReadPages
 from connector_builder.generated.models.stream_read_request_body import StreamReadRequestBody
+from connector_builder.generated.models.stream_read_slices import StreamReadSlices
 from connector_builder.generated.models.streams_list_read import StreamsListRead
+from connector_builder.generated.models.streams_list_read_streams import StreamsListReadStreams
 from connector_builder.generated.models.streams_list_request_body import StreamsListRequestBody
+from connector_builder.impl.low_code_cdk_adapter import LowCodeSourceAdapter
 from fastapi import Body
+from jsonschema import ValidationError
 
 
 class DefaultApiImpl(DefaultApi):
+    def __init__(self):
+        self.logger = logging.getLogger("airbyte.connector-builder")
+
     async def get_manifest_template(self) -> str:
         return "Hello World"
 
     async def list_streams(self, streams_list_request_body: StreamsListRequestBody = Body(None, description="")) -> StreamsListRead:
-        raise Exception("not yet implemented")
+        """
+        Takes in a low code manifest and a config to resolve the list of streams that are available for testing
+        :param streams_list_request_body: Input parameters to retrieve the list of available streams
+        :return: Stream objects made up of a stream name and the HTTP URL it will send requests to
+        """
+        adapter = LowCodeSourceAdapter(streams_list_request_body.manifest)
+
+        stream_list_read = []
+        for http_stream in adapter.get_http_streams(streams_list_request_body.config):
+            stream_list_read.append(
+                StreamsListReadStreams(
+                    name=http_stream.name,
+                    url=urljoin(http_stream.url_base, http_stream.path()),
+                )
+            )
+        return StreamsListRead(streams=stream_list_read)
 
     async def read_stream(self, stream_read_request_body: StreamReadRequestBody = Body(None, description="")) -> StreamRead:
-        raise Exception("not yet implemented")
+        """
+        Using the provided manifest and config, invokes a sync for the specified stream and returns groups of Airbyte messages
+        that are produced during the read operation
+        :param stream_read_request_body: Input parameters to trigger the read operation for a stream
+        :return: Airbyte record messages produced by the sync grouped by slice and page
+        """
+        return self.temp_read_stream(stream_read_request_body)
+
+    def temp_read_stream(self, stream_read_request_body: StreamReadRequestBody = Body(None, description="")) -> StreamRead:
```

Comment:
why is this called temp ?

Reply:
heh because it made it easier for me to test functionality quicker w/o starting the server, but i'll be putting the contents of the method into `read_stream()` once it's out of draft mode.